### DISCUSSION TITLE
Transformer optimization on DirectML not fully working?

### DISCUSSION THREAD

**Discussion by kazssym** [microsoft/onnxruntime#17409] [Other Q&A]
I am using onnxruntime-directml to convert Stable Diffusion models for Olive. Regarding the following line, is DMLExecutionProvider not ready yet or not suitable for optimization in onnxruntime?

https://github.com/microsoft/onnxruntime/blob/6ea3908db482d4e62e16d9f2285928c0b9e48d2c/onnxruntime/python/tools/transformers/optimizer.py#L93

**Comment by skottmckay** [comment]
DMLExecutionProvider is ready and stable. 

That script is for optimizing BERT models and I'd guess that when that was being created/tested it was being used on Linux so DML wasn't included in the list. Not sure how that's related to Olive though.

**Reply by kazssym** [reply]
Thanks.
